.github/workflows/check-code-quality.yml

```diff
@@ -75,3 +75,17 @@ jobs:
             git status --porcelain --untracked-files=no
             exit 1
           fi
+      
+      - name: Check generated code
+        run: |
+          status_before=$(git status --porcelain)
```

Comment:
it looks weird but i understand what it does. however, it looks a little bit weird.

alternatively, you may run this step only `on: success` (please check the github actions documentation for how to do that). then you know for sure that the diff was clean before this check started to run. Then you don't need the multiple `git status`es.

Reply:
@mgovers I see. Using the fact that if the `If needed raise error` job ran successfully, the directory ought to be clean.